devicetypes/qubino/qubino-flush-dimmer-0-10V.src/qubino-flush-dimmer-0-10V.groovy

```diff
@@ -0,0 +1,731 @@
+			state.currentPreferencesState."$it.key".status = "syncPending"
+			if (it.type == "boolRange") {
+				def preferenceName = it.key + "Boolean"
+				if (notNullCheck(settings."$preferenceName")) {
+					if (!settings."$preferenceName") {
+						state.currentPreferencesState."$it.key".status = "disablePending"
+					} else if (state.currentPreferencesState."$it.key".status == "disabled") {
+						state.currentPreferencesState."$it.key".status = "syncPending"
+					}
+				} else {
+					state.currentPreferencesState."$it.key".status = "syncPending"
+				}
+			}
+		} else if (!state.currentPreferencesState."$it.key".value) {
+			log.warn "Preference ${it.key} no. ${it.parameterNumber} has no value. Please check preference declaration for errors."
+		}
+	}
+	syncConfiguration()
+	// Preferences template end
+}
+
+private syncConfiguration() {
+	def commands = []
+	parameterMap.each {
+		if (state.currentPreferencesState."$it.key".status == "syncPending") {
+			commands += encap(zwave.configurationV2.configurationSet(scaledConfigurationValue: getCommandValue(it), parameterNumber: it.parameterNumber, size: it.size))
+			commands += encap(zwave.configurationV2.configurationGet(parameterNumber: it.parameterNumber))
+		} else if (state.currentPreferencesState."$it.key".status == "disablePending") {
+			commands += encap(zwave.configurationV2.configurationSet(scaledConfigurationValue: it.disableValue, parameterNumber: it.parameterNumber, size: it.size))
+			commands += encap(zwave.configurationV2.configurationGet(parameterNumber: it.parameterNumber))
+		}
+	}
+	sendHubCommand(commands)
+}
+
+def getCommandClassVersions() {
+	[
+		0x20:1, // Basic
+		0x25:1, // Switch Binary
+		0x26:3, // Switch Multilevel
+		0x27:1, // Switch All
+		0x59:1, // Association Grp Info
+		0x5A:1, // Device Reset Locally
+		0x70:2, // Configuration
+		0x72:2, // Manufacturer Specific
+		0x73:1, // Powerlevel
+		0x85:2, // Association
+		0x86:1, // Version
+		0x8E:2  // Multi Channel Association
+	]
+}
+
+def configure() {
+	def commands = []
+
+	commands << zwave.manufacturerSpecificV2.manufacturerSpecificGet().format()
+	commands << zwave.associationV1.associationRemove(groupingIdentifier:1).format()
+	commands << zwave.associationV1.associationRemove(groupingIdentifier:2).format()
+	commands << zwave.associationV1.associationRemove(groupingIdentifier:3).format()
+	commands << zwave.associationV1.associationRemove(groupingIdentifier:4).format()
+	commands << zwave.associationV1.associationSet(groupingIdentifier:1, nodeId:[zwaveHubNodeId]).format()
+	commands << zwave.associationV1.associationSet(groupingIdentifier:2, nodeId:[zwaveHubNodeId]).format()
+	commands << zwave.associationV1.associationSet(groupingIdentifier:3, nodeId:[zwaveHubNodeId]).format()
+	commands << zwave.associationV1.associationSet(groupingIdentifier:4, nodeId:[zwaveHubNodeId]).format()
```

Comment:
can you explain what's going on here?

Reply:
@greens 
There probably should be **nodeId:[zwaveHubNodeId]** added to associationRemove calls.
Please correct me if i'm wrong -> the intention was to remove the hub from those association groups and then add it again - to avoid doubling hubNodeId in device's association table.
If device deals with it on the protocol level, then that associationRemove call are redundant and they will be removed from the code.

Reply:
Removing a device from the network should clear its association table.

I mostly just wanted a comment in the code to explain this non-standard code, but it sounds like the removes aren't necessary.